src/partner_works/entities/partner_site.clj

```
@@ -96,53 +96,39 @@
+  tx-data. If there are no retractions it returns an empty vector."
+  [current-eid component-ky components]
+  (if (and current-eid (seq components))
     (mapv
-     #(vector :db/retract ps-eid ky (:db/id %))
+     #(vector :db/retract current-eid component-ky (:db/id %))
      components)
     []))
 
-(defn customization-tx-data
-  "For a given current partner-site and a new partner-site returns a
-  vector of additions and retractions to the :partner/customization
-  ref and the customizations themselves formatted as a list of maps
-  (as tx-data, as they are passed in here)."
-  [{customizations :partner/customization ps-eid :db/id :as current-partner-site}
-   {new-customizations :partner/customization new-ps-eid :db/id :as new-partner-site}]
-  (-> (component-additions ps-eid new-ps-eid :partner/customization
-                           new-customizations)
-      (into (component-retractions ps-eid :partner/customization
-                                   customizations))))
-
-(defn alternate-hostname-tx-data
-  "Returns a vector containing additions and retractions for
-  alternate-hostnames."
-  [{alternate-hostnames :partner/alternate-hostname
-    ps-eid :db/id :as current-partner-site}
-   {new-alternate-hostnames :partner/alternate-hostname
-    new-ps-eid :db/id :as new-partner-site}]
-  (-> (component-additions ps-eid new-ps-eid :partner/alternate-hostname
-                           new-alternate-hostnames)
-      (into (component-retractions ps-eid :partner/alternate-hostname
-                                   alternate-hostnames))))
+(defn component-tx-data
+  "For a current parent entity and a new parent-entity returns a
+  vector of additions and retractions to the component at
+  component-ky. The return is a list of maps of tx-data."
+  [{current-eid :db/id :as current-entity}
+   {new-eid :db/id :as new-entity} component-ky]
+  (let [components (get current-entity component-ky)
```

So this is quite a digression at this point, but: if we have something that we know is going to be called explicitly in the way you've written it, then we shouldn't have to worry that it is going to be nil, and if it IS going to be nil we should let it blow up and figure out why the hell it is nil.  And if you DO expect nils, then you should have an alternate value provided in your `get` call.  Don't think this makes a lot of sense otherwise.

But again, a quibble. Also goes to show yet another reason why `nil` being a member of every type is evil: you never know when it's going to rear its ugly head and start encoding that fear into stylistic decisions across a codebase...